### DISCUSSION TITLE
Dilithium 5 issue in 32-bit

### DISCUSSION THREAD

**Discussion by Crawl12321** [open-quantum-safe/liboqs#1628] [Q&A] [answered]
Hello,

During our integration of libOQS, we’ve stumbled on an issue with DILITHIUM5 (DILITHIUM 2 and 3 are working just fine).
The situation is the following:
-	We are using several CoreI3/5/7 64-bit processors of different generations, some old and some new
-	On these processors, we put a 32-bit Linux OS (kernel 5.15) built with buildroot
-	We’ve successfully integrated libOQS in our Linux (at first on a 3rd-generation CoreI7)
-	We can call DILITHIUM2 and 3 related cryptographic functions (key-pair generation, signature, verification) without any issue
-	We can call DILITHIUM5 key-pair generation, without any issue
-	When we can call DILITHIUM5 signature, there is a segmentation fault in libOQS
Would you, by any chance, see a reason for this issue? Is libOQS fully 32-bit compatible? Could there be a correlation with the processors hardware features?

**Comment by baentsch** [comment]
> Is libOQS fully 32-bit compatible?  Could there be a correlation with the processors hardware features?

Most definitely you could have hit on a hardware support "snag": The hardware/platform support status of `liboqs` is [under discussion](https://github.com/open-quantum-safe/liboqs/pull/1605), so I personally would not vouch for all 32bitters right now. What you should do, though, is [build/use `liboqs` without any hardware optimization](https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#OQS_DIST_BUILD) thus activating reference code that _should_ work everywhere. If it does not, please open a [complete bug report](https://github.com/open-quantum-safe/liboqs/issues/new?assignees=&labels=&projects=&template=bug_report.md&title=).

**Comment by Crawl12321** [comment, marked as answer]
Thank you for your answer @baentsch  :)

We stumbled on the root of our issue while trying to make an exemple, the dilithium5 signature process was exceeding the thread stack size. Problem solved.
